src/formatDate.js

```diff
@@ -51,6 +51,35 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const newDate = [];
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+
+  const arrDate = date.split(oldSeparator);
+
+  const objDate = {};
+
+  for (let i = 0; i <= 2; i++) {
+    objDate[fromFormat[i]] = arrDate[i];
+  }
+
+  if (objDate['YY'] < 30) {
+    objDate['YYYY'] = 20 + objDate['YY'];
+  }
+
+  if (objDate['YY'] >= 30) {
+    objDate['YYYY'] = 19 + objDate['YY'];
+  }
+
+  if (objDate['YYYY'] !== undefined) {
+    objDate['YY'] = objDate['YYYY'].slice(2, 4);
+  }
```

Comment:
This part will rewrite your existed property with the same value. I'll explain: before this lines you check if (oldDateObject['YY']) exists and if true create oldDateObject['YYYY']. Then here you check if (oldDateObject['YYYY']) exists - of course it exists as you've already created it.
You need to improve the logic of these blocks to avoid unnecessary runs.

Simply use else statement.